tools/clang/lib/CodeGen/CGHLSLMSFinishCodeGen.cpp

```diff
@@ -1530,7 +1531,22 @@ Value *TryEvalIntrinsic(CallInst *CI, IntrinsicOp intriOp) {
     return EvalUnaryIntrinsic(CI, floorf, floor);
   } break;
   case IntrinsicOp::IOP_round: {
-    return EvalUnaryIntrinsic(CI, roundf, round);
+    // Both FXC and DXC would apply nearest even rounding mode on variables, but on compile-time constants
+    // it would apply away from zero for midway values. This would cause differences in results in cases such
+    // as round(x) (where x is 0.5) and round(0.5) where the former would evaluate to 1 and latter would
+    // evaluate to 0.
+    //
+    // For compatibility with FXC,  DXC still preserve above behavior for language version 2016 or below.
+    // However for newer language version, DXC would always use nearest even for round() intrinsic in all
```

Comment:
just trying to make sure I understand. I think "would" here should be "should"?

Reply:
updated.